minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterImpl.java

```diff
@@ -666,27 +666,34 @@ private void verifyUp() throws InterruptedException, IOException {
             instanceId = new String(bytes, UTF_8);
             break;
           } catch (KeeperException e) {
+            log.warn("Error trying to read instance id from zookeeper: " + e.getMessage());
             log.debug("Unable to read instance id from zookeeper.", e);
           }
         }
         Thread.sleep(1000);
       }
 
       if (instanceId == null) {
-        try {
-          log.warn("******* COULD NOT FIND INSTANCE ID - DUMPING ZK ************");
-          log.warn("Connected to ZooKeeper: {}", getZooKeepers());
-          log.warn("Looking for instanceId at {}",
-              Constants.ZROOT + Constants.ZINSTANCES + "/" + config.getInstanceName());
-          ZKUtil.visitSubTreeDFS(zk, Constants.ZROOT, false, new StringCallback() {
-            @Override
-            public void processResult(int rc, String path, Object ctx, String name) {
-              log.warn("{}", path);
+        for (int i = 0; i < numTries; i++) {
+          if (zk.getState().equals(States.CONNECTED)) {
```

Comment:
When comparing equality with enums, you should always use `==`, not `.equals` so you get compile-time type safety. I'll submit a follow-up commit a fix to this, since it's already merged.

Reply:
:+1: 

I copied it from another place in the code, so there are at least two places in the code that need to be fixed

Reply:
Yeah, I got that one, too. :smiley_cat: